astropy/cosmology/io/mrt.py

```diff
@@ -0,0 +1,116 @@
+        the cosmology instance. The class also provides default parameter
+        values, filling in any non-mandatory arguments missing in 'table'.
+    **kwargs : Any
+        Additional keyword arguments passed to :attr:`astropy.table.QTable.read`.
+        The 'format' is set to 'mrt', regardless of input.
+
+    Returns
+    -------
+    |Cosmology| subclass instance
+
+    Raises
+    ------
+    ValueError
+        If the keyword argument 'format' is given and is not "mrt".
+    """
+    # Check that the format is 'mrt' (or not specified)
+    format = kwargs.pop("format", "mrt")
+    if format != "mrt":
+        raise ValueError(f"format must be 'mrt', not {format}")
+
+    # Reading is handled by `QTable`.
+    table = QTable.read(filename, format="mrt", **kwargs)
+
+    # Create a single column named 'm_nu' by combining the values from the 'mnu0', 'mnu1', and 'mnu2' columns
+    m_nu_data = []
+    for i in ("mnu0", "mnu1", "mnu2"):
+        column_data = table[i][0]
+        m_nu_data.append(column_data.value)
+        table.remove_column(i)
+
+    table.add_column([m_nu_data], name="m_nu", index=-2)
+
+    # Build the cosmology from table, using the private backend.
+    return from_table(
+        table, index=index, move_to_meta=move_to_meta, cosmology=cosmology
+    )
+
+
+def write_mrt(cosmology, file, *, overwrite=False, cls=QTable, **kwargs):
+    """Serialize the |Cosmology| into an MRT file.
+
+    Parameters
+    ----------
+    cosmology : |Cosmology| subclass instance
+        The |Cosmology| object to serialize into an MRT file.
+    file : path-like or file-like
+        The file path or file-like object where the MRT file will be written.
+    overwrite : bool, optional keyword-only
+        Whether to overwrite the file if it already exists.
+    cls : |Table| class, optional keyword-only
+        The |Table| class to use when writing the MRT file. Default is |QTable|.
+    **kwargs : Any
+        Additional keyword arguments passed to ``cls.write``.
+
+    Raises
+    ------
+    TypeError
+        If kwarg (optional) 'cls' is not a subclass of `astropy.table.Table`
+    ValueError
+        If the keyword argument 'format' is given and is not "mrt".
+    """
+    # Check that the format is 'mrt' (or not specified)
+    format = kwargs.pop("format", "mrt")
+    if format != "mrt":
+        raise ValueError(f"format must be 'mrt', not {format}")
+
+    # Set cosmology_in_meta as false for now since there is no metadata being kept
+    table_main = to_table(cosmology, cls=cls, cosmology_in_meta=False)
+    table = represent_mixins_as_columns(table_main)
+
+    # Replace the m_nu column with three columns with names 'mnu0', 'mnu1', 'mnu2'
+    m_nu = table_main["m_nu"]
+    table.remove_column("m_nu")
+    for i in range(len(m_nu[0])):
+        column_name = f"mnu{i}"
+        column_data = m_nu[0][i]
+        table.add_column(column_data, name=column_name, index=-2)
```

Comment:
With `represent_mixins_as_columns(table_main)`, does this serialize without breaking apart the neutrino mass?

Reply:
I may be completely wrong in my understanding of how serializing is supposed to work in this context. But I believe ```represent_mixins_as_columns(table_main)```, does not serialize the table without breaking apart the neutrino mass, and here is why:

```python
from astropy.cosmology import Planck18
from astropy.table.serialize import represent_mixins_as_columns
table_main = Planck18.to_format("astropy.table")
table = represent_mixins_as_columns(table_main)

m_nu_table_main = table_main['m_nu'] # Get the m_nu column from table_main
m_nu_table = table['m_nu']  # Get the m_nu column from table

N = len(m_nu_table[0])  # Get the number of elements in m_nu

for i in range(N):
    column_data_table = m_nu_table[0][i]
    print(column_data_table, type(column_data_table))
    
for i in range(N):
    column_data_table_main = m_nu_table_main[0][i]
    print(column_data_table_main, type(column_data_table_main))
```
Output:
```plaintext
0.0 <class 'numpy.float64'>
0.0 <class 'numpy.float64'>
0.06 <class 'numpy.float64'>
0.0 eV <class 'astropy.units.quantity.Quantity'>
0.0 eV <class 'astropy.units.quantity.Quantity'>
0.06 eV <class 'astropy.units.quantity.Quantity'>
```
The way I see it is that, for serialization to work correctly the outputs must be the same.

Reply:
You're right that `represent_mixins_as_columns` doesn't help here. I'll open an issue to see if we can extend `represent_mixins_as_columns` to make structured arrays into a series of columns.

Reply:
ping @taldcroft

Reply:
I opened https://github.com/astropy/astropy/issues/14913 for this change.